nbsite/nbbuild.py

```diff
@@ -147,31 +152,74 @@ def preprocess(self, nb, resources):
         nbc.cells = nbc.cells[start:end]
         return nbc, resources
 
-    def __call__(self, nb, resources): return self.preprocess(nb,resources)
+    def __call__(self, nb, resources):
+        return self.preprocess(nb,resources)
 
 
-def comment_out_details(source):
+class FixNotebookLinks(Preprocessor):
     """
-    Given the source of a cell, comment out  any lines that contain <details>
+    Fixes relative notebook links by pointing to ReST or Markdown
+    source files and if necessary stripping leading numbering
+    (e.g. 0_notebook.ipynb) which are stripped by the generate-rst
+    script.
     """
```

Comment:
Lots of regular expressions in this class! Hope I won't ever need to touch it ;-p

Reply:
Mostly stolen from the `fix_links` script, so not really anything new. We now have to fix links before we render because Sphinx now actually parses the links and will mangle them if they're not valid.